src/components/Charts/Axes/Axes.js

```diff
@@ -36,7 +42,23 @@ export default class Axes extends React.Component {
     const fontSize = parseInt(ref?.style?.fontSize, 10)
     return isNaN(fontSize) ? 0 : fontSize
   }
+  componentDidUpdate = (prevProps, prevState) => {
+    if (prevState.currentRowNumber !== this.state.currentRowNumber) {
+      this.forceUpdate()
+    }
+  }
 
+  renderLoadMoreDropdown = (currentRowNumber, totalRowNumber) => {
+    return (
+      <tspan id={`load-more-drop-down-span-${this.COMPONENT_ID}`}>
+        <tspan id={`visualizing-span-${this.COMPONENT_ID}`}>{`Visualizing `}</tspan>
```

Comment:
I see the unique identifier here.. should you remove that variable you set in the constructor?

Reply:
ideally we should use refs as much as we can instead of document.querySelector. We can make a jira for this to do later though as this feature is high priority for nike